src/components/BookForm.js

```diff
@@ -1,24 +1,31 @@
 import React from 'react';
 import { v4 as uuidv4 } from 'uuid';
+import axios from 'axios';
 import { useDispatch } from 'react-redux';
-import { added } from '../Redux/books/booksSlice';
+import { getBooks } from '../Redux/books/booksSlice';
 
 const BookForm = () => {
   const dispatch = useDispatch();
-
   const handleSubmit = (e) => {
     e.preventDefault();
 
+    const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/IfJm8IrVNYBvy5mLAJXh/books';
+
     const newBook = {
-      id: uuidv4(),
+      item_id: uuidv4(),
       title: e.target.title.value,
       category: e.target.category.value,
       author: e.target.author.value,
     };
 
-    dispatch(added(newBook));
-
-    e.target.reset();
+    axios.post(url, newBook)
+      .then((response) => {
+        console.log('Response:', response.data);
+        dispatch(getBooks());
+      })
+      .catch((error) => {
+        console.error('Failed to create a new book:', error);
+      });
```

Comment:
> - [ ] Great job, @Xghayor! Could you please eliminate the usage of console.log and console.error from your code? These statements are contributing to the untidiness of your code. 😞
> 
> Also, I suggest you place the Axios POST request directly in the redux bookSlice, as it follows the recommended pattern for handling asynchronous operations with Redux. Keeping API calls and state management logic separate is a good practice. With `createAsyncThunk`, you encapsulate the API call logic in the Redux action creator rather than scattering it throughout your components.😊

```
export const addBook = createAsyncThunk(
  'books/addBook',
  async (bookData) => {
    try {
      const response = await axios.post(url, bookData);
      return response.data;
    } catch (err) {
      throw err;
    }
  },
);
```